src/elements/content-sidebar/activity-feed/__tests__/Avatar-test.js

```diff
@@ -9,66 +9,57 @@ describe('elements/content-sidebar/ActivityFeed/Avatar', () => {
         name: 'foo bar',
         type: 'user',
     };
-    const getAvatarUrl = jest.fn().mockReturnValue(Promise.resolve('foo'));
 
     const getWrapper = props => shallow(<Avatar {...props} />);
 
-    test('should render nothing if no avatarUrl in state and getAvatarUrl method was passed', () => {
-        expect(
-            getWrapper({ user, getAvatarUrl })
-                .dive()
-                .find('Avatar')
-                .exists(),
-        ).toBe(false);
-        expect(getAvatarUrl).toBeCalledWith(user.id);
-    });
-
-    test('should render avatar with initials if getAvatarUrl is not passed in and no avatarUrl is in state', () => {
-        expect(
-            getWrapper({ user })
-                .find('AvatarInitials')
-                .exists(),
-        ).toBe(false);
-        expect(getAvatarUrl).not.toBeCalledWith(user.id);
+    test('should pass isPending prop to wrapped Avatar getAvatarUrl prop is passed', () => {
+        const getAvatarUrl = jest.fn().mockResolvedValue('avatar.jpg');
+        const wrapper = getWrapper({ user, getAvatarUrl });
+        const wrappedAvatar = wrapper.find('Avatar').first();
+        expect(wrappedAvatar.props()).toEqual(expect.objectContaining({ isPending: true }));
+        expect(wrapper).toMatchInlineSnapshot(`
+<Avatar
+  avatarUrl={null}
+  id="foo"
+  isPending={true}
+  name="foo bar"
+/>
+`);
     });
 
-    test('should render the avatar with an avatarUrl', () => {
+    test('should call getAvatarUrl prop with user id and pass result to child', async () => {
+        const getAvatarUrl = jest.fn().mockResolvedValue('avatar.jpg');
         const wrapper = getWrapper({ user, getAvatarUrl });
-        wrapper.instance().getAvatarUrlHandler('foo');
-        wrapper.update();
-        expect(wrapper).toMatchSnapshot();
+        expect(getAvatarUrl).toBeCalledWith(user.id);
+        // wait a tick for async, use react-dom/test-utils/act in future versions
+        await new Promise(r => setImmediate(r));
+        const wrappedAvatar = wrapper.find('Avatar').first();
+        expect(wrappedAvatar.props()).toEqual(expect.objectContaining({ isPending: false, avatarUrl: 'avatar.jpg' }));
+        expect(wrapper).toMatchInlineSnapshot(`
```

Comment:
We don't really use these much, it looks like. Can this live in a snapshot file?